source/engine/graphics/GpuAllocator.h

```diff
@@ -48,7 +49,7 @@ namespace graphics
     class GpuAllocator
     {
         public:
-            GpuAllocator( vk::Device logicalDevice, vk::PhysicalDevice physicalDevice, vk::Instance instance, Commands* commands);
+            GpuAllocator(const Device* device, const Instance& instance);
```

Comment:
Why const ptr here?

Reply:
As opposed to non-const pointer, or const reference? Its const because it can be; its a pointer because `GpuAllocator` keeps it as a member. Our current convention is that we don't store references as members. Mostly that's because it gets in the way of move operations, which doesn't really apply to this non-moveable type, but I figured why break the convention?